src/collections/Form/Form.js

```diff
@@ -7,8 +8,22 @@ export default class Form extends Component {
   static propTypes = {
     children: PropTypes.node,
     className: PropTypes.string,
+    settings: PropTypes.object,
   };
 
+  componentDidMount() {
+    this.element = $(this.refs.element);
+    this.element.form(this.props.settings);
+  }
+
+  componentWillUnmount() {
+    this.element.off();
+  }
+
+  plugin() {
+    return this.element.form(...arguments);
+  }
+
```

Comment:
There may be some listeners getting away from clean-up, so we'll want to keep an eye on that. To reproduce start memory profiling and then work the form to completion and click submit.